api/sso.go

```diff
@@ -110,7 +115,13 @@ func (a *API) SingleSignOn(w http.ResponseWriter, r *http.Request) error {
 		return internalServerError("Error creating SAML authentication request redirect URL").WithInternalError(err)
 	}
 
-	http.Redirect(w, r, ssoRedirectURL.String(), http.StatusSeeOther)
+	if params.UseHTTPRedirect {
+		http.Redirect(w, r, ssoRedirectURL.String(), http.StatusSeeOther)
+	} else {
+		return sendJSON(w, http.StatusOK, SingleSignOnResponse{
+			URL: ssoRedirectURL.String(),
+		})
+	}
```

Comment:
can we default to `use_http_redirect` instead? this would be identical to the `GET /authorize` endpoint

Reply:
Yes similar, tho SAML requests will benefit a lot from CAPTCHA since each time you attempt logging in a new entry is saved in the database (the `saml_relay_states` table) because the RelayState parameter echoed back from the Identity Provider (similar to the `state=` parameter in OAuth) is limited to 86 chars. So we have to store the data in the database. POSTs are also better because we can send quite a bit of arbitrary data in the future without having it all appear in the URL and in access logs.